field/composite.go

```diff
@@ -342,6 +338,20 @@ func (f *Composite) UnmarshalJSON(b []byte) error {
 }
 
 func (f *Composite) pack() ([]byte, error) {
+
+	// getting max length of data
+	encodedLength, err := f.spec.Pref.EncodeLength(f.spec.Length, f.spec.Length)
```

Comment:
Is this right?

Reply:
yes
added comment for this

the code is to get real data size without length prefix
length prefix of variable field is formatted string with same size, L/LL/LLL
so we can get real size of length prefix from spec length (instead of len(packed))